src/ai/backend/manager/event_dispatcher/dispatch.py

```diff
@@ -178,7 +191,7 @@ def _dispatch_error_monitor_events(self, event_dispatcher: EventDispatcher) -> N
         evd.consume(
             AgentErrorEvent,
             None,
-            self._event_dispatcher_plugin_ctx.handle_event,
+            self._plugin_event_handler.handle_event,
```

Comment:
Rather than focusing on the handling logic itself, it might be better to focus on the event.
Wouldn’t it be better to handle this together in the agent_event_handler? @fregataa

Reply:
I divided them into different dispatch functions because the handler gets Session execution events, which are not agent events.